transport/httprp/server_test.go

```diff
@@ -0,0 +1,115 @@
+package httprp_test
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	httptransport "github.com/go-kit/kit/transport/httprp"
+)
+
+func TestServerHappyPathSingleServer(t *testing.T) {
+	originServer := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("hey"))
+		}))
+	defer originServer.Close()
+	originURL, _ := url.Parse(originServer.URL)
+
+	handler := httptransport.NewServer(
+		context.Background(),
+		originURL,
+	)
+	proxyServer := httptest.NewServer(handler)
+	defer proxyServer.Close()
+
+	resp, _ := http.Get(proxyServer.URL)
+	if want, have := http.StatusOK, resp.StatusCode; want != have {
+		t.Errorf("want %d, have %d", want, have)
+	}
+
+	responseBody, _ := ioutil.ReadAll(resp.Body)
+	if want, have := "hey", string(responseBody); want != have {
+		t.Errorf("want %d, have %d", want, have)
+	}
+}
+
+func TestServerHappyPathSingleServerWithServerOptions(t *testing.T) {
+	originServer := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if want, have := "go-kit-proxy", r.Header.Get("X-TEST-HEADER"); want != have {
+				t.Errorf("want %d, have %d", want, have)
+			}
+
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("hey"))
+		}))
+	defer originServer.Close()
+	originURL, _ := url.Parse(originServer.URL)
+
+	handler := httptransport.NewServer(
+		context.Background(),
+		originURL,
+		httptransport.ServerBefore(func(ctx context.Context, r *http.Request) context.Context {
+			r.Header.Add("X-TEST-HEADER", "go-kit-proxy")
```

Comment:
Would you mind extracting these strings to constants in the test function scope?

```go
const (
    headerKey = "X-TEST-HEADER"
    headerVal = "go-kit-proxy"
)
```